source.go

```diff
@@ -106,6 +113,11 @@ func (s *Source) Read(ctx context.Context) (sdk.Record, error) {
 
 func (s *Source) Ack(ctx context.Context, position sdk.Position) error {
 	sdk.Logger(ctx).Debug().Str("position", string(position)).Msg("got ack")
+	index, err := s.indexQueue.Dequeue()
+	if err != nil {
+		return err
```

Comment:
Maybe wrap the error and add a description what's going on.

Reply:
this shouldn't even happen, would the description be something blaming conduit that something wrong happened with it?

Reply:
I would just describe what we know, that it's likely a bug, that we were not expecting to get an ack because all records sent to conduit already received an ack.